### DISCUSSION TITLE
Allow unassign self via REST API

### DISCUSSION THREAD

**Discussion by tgmpje** [community/community#15650] [Projects and Issues]
If an issue is assigned to you, you can always unassign yourself by simply clicking "unassign me" (which performs an AJAX request to `https://github.com/{owner}/{repo}/issues/{issue_number}/unassign_self`). You do not need to have any special rights in the repository to do this.

Unfortunately there is no equivalent REST API call to do this.
The https://docs.github.com/en/rest/issues/assignees#remove-assignees-from-an-issue (`DELETE https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}/assignees`) can only be used whilst having admin rights.

It would be great to have something like `DELETE https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}/unassign_self`, or be able to use `DELETE https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}/assignees` to unassign yourself (without needing admin rights)

**Comment by duaneking** [comment]
The problem is that we don't always get the option to unassign ourselves.  You do seem to have to have special permissions in order to be able to remove yourself as somebody in that repository assigns you work without your consent and that just makes GitHub a worse place to be.
